Engine/AlgorithmManager.cs

```diff
@@ -545,7 +545,8 @@ public void Run(AlgorithmNodePacket job, IAlgorithm algorithm, ISynchronizer syn
                         var timeKeeper = algorithm.TimeKeeper;
                         foreach (var update in timeSlice.ConsolidatorUpdateData)
                         {
-                            var consolidators = update.Target.Consolidators;
+                            // Creates a list from the hashset to enable item removal in the IDataConsolidator.Update callback
+                            var consolidators = update.Target.Consolidators.ToList();
```

Comment:
I feel like this _shouldn't_ be required... and if a callback is causing modification of a list that raises an exception _somewhere_ than I suspect we're doing something else wrong and this may just be a bandaid.